### PR TITLE
add check/uncheck

### PR DESCRIPTION
**Pull request by DIFSRIP**
- Add requests function to check/uncheck select boxes (ui5 / nonui5)
- Solves issue with limited type for fill.

### CODE REVIEW

File: src/reuse/modules/nonUi5/element.ts [hunk 1]
```diff
@@ -423,6 +423,20 @@ export class ElementModule {
...
+   * @memberOf nonUi5.element
+   * @description Returns a boolean if the element (e.g. checkbox) is selected.
+   * @param {Object} elem - The element.
+   * @returns {boolean}
+   * @example const elem = await nonUi5.element.getById("elem01");
+   * const isSelected = await nonUi5.element.isSelected(elem);
+   */
+  async isSelected(elem: Element): Promise<boolean> {
+    const vl = this.vlf.initLog(this.isSelected);
+    return elem.isSelected();
```
Comment (cvakiitho) [inline src/reuse/modules/nonUi5/element.ts:437]: Why are we wrapping wdio element method, for better visibility to teams?

File: src/reuse/modules/common/userInteraction.ts [hunk 2]
```diff
@@ -15,28 +15,29 @@ export class UserInteraction {
...
+
+    if (typeof value === "number" || typeof value === "string" || typeof value === "string") {
       try {
         const elem = await $(await browser.getActiveElement());
         await elem.addValue(value);
       } catch (error) {
         throw new Error(`Function 'fillActive' failed: ${error}`);
       }
+    } else {
+      throw new Error("Function 'fillActive' failed: Please provide a value(datatype - number/string) as argument.");
```
Comment (cvakiitho) [inline src/reuse/modules/common/userInteraction.ts:32]: I'd expect TS to catch this in compiler, but this is probably a fallback for JS tests right?

File: src/reuse/modules/common/userInteraction.ts [hunk 3]
```diff
@@ -15,28 +15,29 @@ export class UserInteraction {
...
    * @example await common.userInteraction.fillActive("My Value");
    */
-  async fillActive(value: string) {
+  async fillActive(value: string | number | boolean) {
     const vlf = this.vlf.initLog(this.fillActive);
-    if (!(typeof value === "number" || typeof value === "string")) {
-      throw new Error("Function 'fillActive' failed: Please provide a value(datatype - number/string) as argument.");
-    } else {
+
+    if (typeof value === "number" || typeof value === "string" || typeof value === "string") {
```
Comment (cvakiitho) [inline src/reuse/modules/common/userInteraction.ts:24]: duplicate string here - but as previous comment - it probably can only be number/string?

File: src/reuse/modules/nonUi5/element.ts (hunk 1 above)
Comment (DIFSRIP) [reply src/reuse/modules/nonUi5/element.ts:437]: - we wrap everything, to be able to keep compatibility (same as moving from vyper to qmate, or if wdio does changes)
- to keep consistent usage (we have function where we pass the elem as argument, wdio has functions on elements)
- to provide flexibility and additional checks (in this case only the vl)
- transparency through Reuse API

File: src/reuse/modules/common/userInteraction.ts (hunk 2 above)
Comment (DIFSRIP) [reply src/reuse/modules/common/userInteraction.ts:32]: sure, most tests are written in js, so we need to provide proper error handling.

File: src/reuse/modules/common/userInteraction.ts [hunk 4]
```diff
@@ -15,28 +15,29 @@ export class UserInteraction {
    * @function fillActive
    * @memberOf common.userInteraction
    * @description Fills the active input with the given value.
-   * @param {String} value - The value with witch the input should be filled.
+   * @param {String | Number | Boolean} value - The value to enter.
    * @example await common.userInteraction.fillActive("My Value");
    */
-  async fillActive(value: string) {
+  async fillActive(value: string | number | boolean) {
```
[the remark this answers is not in the document]
Comment (DIFSRIP) [reply src/reuse/modules/common/userInteraction.ts:21]: I know, strangely SIV team was using this before with boolean value to check/uncheck checkboxes. @PhyNerd requested to support this but I am not sure if and how it works.

File: src/reuse/modules/common/userInteraction.ts (hunk 3 above)
Comment (DIFSRIP) [reply src/reuse/modules/common/userInteraction.ts:24]: thanks will remove the duplicate check

File: src/reuse/modules/nonUi5/userInteraction.ts [hunk 5]
```diff
@@ -219,35 +280,39 @@ export class UserInteraction {
...
    * @memberOf nonUi5.userInteraction
    * @description Clears and fills the passed input element.
    * @param {Object} element - The element.
-   * @param {String} value - The value to be filled in.
+   * @param {String | Number} value - The value to enter in.
    * @example const elem = await nonUi5.element.getById("input01");
    * await nonUi5.userInteraction.clearAndFill(elem, "Service 01");
    */
-  async clearAndFill(element: Element, value: string) {
+  async clearAndFill(element: Element, value: string | string) {
```
Comment (heraldantony) [inline src/reuse/modules/nonUi5/userInteraction.ts:303]: typo - string | number

**Review by heraldantony — APPROVED** [review]
looks good

File: src/reuse/modules/nonUi5/userInteraction.ts (hunk 5 above)
Comment (DIFSRIP) [reply src/reuse/modules/nonUi5/userInteraction.ts]: done

**Review by heraldantony — APPROVED** [review]
looks good

**Pull request merged**